### PR TITLE
Fix a get_ria_vector for FE_BDM

### PR DESCRIPTION
**Pull request by elddle**
The number of `dofs_per_face` in a BDM element is the number of polynomials of `deg+1`, rather than `deg`. The function `get_dpo_vector` actually uses the right number, so I modified this one accordingly.

Also, the assertion was modified, since it does not make sense in the lowest order case where the number of face degrees of freedom is exactly equal to the total number of degrees of freedom per cell, as there are no internal DOFs for BDM_1. It was working before because the number of face DOFs was less than it should have been.

### CODE REVIEW

**Comment by davydden** [comment]
do you mind adding a unit test that checks this?

File: tests/fe/bdm_16.cc [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+// the test is used to check the restriction_is_additive flags
+
+#include "../tests.h"
+#include <deal.II/base/logstream.h>
+#include <deal.II/fe/fe_bdm.h>
+
+#include <fstream>
+#include <string>
+
+#define PRECISION 1
```
Comment (masterleinad) [inline tests/fe/bdm_16.cc]: You can probably can remove this. See below.

**Comment by elddle** [comment]
Sorry, I think I messed up by trying to squash my commits, could you please help with solving it ?

**Comment by davydden** [comment]
@elddle do you have a backup branch? Whatever you did wrong, i don't see your original commits here anymore :-(

**Comment by elddle** [comment]
@davydden I have all the modified files backed up, not a branch though

**Comment by davydden** [comment]
then just do `git reset --hard master` and add those changes again and commit.

**Comment by drwells** [comment]
@elddle Usually, when one is in such circumstances, it is useful to use the `reflog`. For example, if I run `git reflog` right now I see
```
[drwells@archway build]$ git reflog
7e71de387b HEAD@{0}: rebase -i (finish): returning to refs/heads/doxygen-link-fixes
7e71de387b HEAD@{1}: rebase -i (fixup): Fix some dead UNV links.
d59bf075eb HEAD@{2}: rebase -i (start): checkout d59bf075eb^
7247dbfdf8 HEAD@{3}: commit: fixup! Fix some dead UNV links.
d59bf075eb HEAD@{4}: checkout: moving from analogous-grammar-fix to doxygen-link-fixes
2b172e7b9d HEAD@{5}: rebase -i (finish): returning to refs/heads/analogous-grammar-fix
2b172e7b9d HEAD@{6}: rebase -i (pick): Write 'is analogous to', not 'is analogy to'.
5325e5ccf0 HEAD@{7}: rebase -i (start): checkout 1ab924f225^
```

and I can run `git reset --hard HEAD@{6}` to restore the repository to its state at that change.

**Comment by elddle** [comment]
@davydden @drwells Thank you, it seems working now. Sorry for messing this up, my first actual experience with github

File: tests/fe/bdm_16.cc [hunk 2]
```diff
@@ -0,0 +1,63 @@
+// ---------------------------------------------------------------------
+//
+// Copyright (C) 2003 - 2017 by the deal.II authors
```
Comment (davydden) [inline tests/fe/bdm_16.cc]: please adjust years

File: tests/fe/bdm_16.cc [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+// it, and/or modify it under the terms of the GNU Lesser General
+// Public License as published by the Free Software Foundation; either
+// version 2.1 of the License, or (at your option) any later version.
+// The full text of the license can be found in the file LICENSE at
+// the top level of the deal.II distribution.
+//
+// ---------------------------------------------------------------------
+
+
+// the test is used to check the restriction_is_additive flags
```
Comment (davydden) [inline tests/fe/bdm_16.cc]: and the comment

File: tests/fe/bdm_16.cc [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+
+std::ofstream logfile ("output");
+
+template<int dim>
+void
+test (const unsigned int degree)
+{
+  FE_BDM<dim> fe_bdm(degree);
+
+  deallog << "Degree=" << degree
```
Comment (davydden) [inline tests/fe/bdm_16.cc]: i think you was testing `dofs_per_face` ?

Comment (davydden) [inline tests/fe/bdm_16.cc]: or call `get_ria_vector`? I don't know anything about `FE_BDM`, but the unit test seems unrelated to what you fix int he library.

Comment (elddle) [inline tests/fe/bdm_16.cc]: No, I was testing if the get_ria_vector indeed returns a vector of # face_dofs 0s followed by # interior dofs 1s, as the face DoFs must have continuity while the interior ones must not

File: tests/fe/bdm_16.cc (hunk 2 above)
Comment (elddle) [inline tests/fe/bdm_16.cc]: Which ones should I use?

File: tests/fe/bdm_16.cc (hunk 4 above)
Comment (davydden) [inline tests/fe/bdm_16.cc]: oh, i see. Then please elaborate it in the in-code comment at the beginning of the test file so that if someone looks at this file in a few years, it would be easier to understand why the test is here.

File: tests/fe/bdm_16.cc (hunk 2 above)
Comment (davydden) [inline tests/fe/bdm_16.cc]: since it's `2017` today and you contribute this file today, i would say change `2003 - 2017` to `2017` :smile:

Comment (elddle) [inline tests/fe/bdm_16.cc]: Ah, ok, I was just trying to be consistent with the other files' copyright formats, I will change it 😄

Comment (davydden) [inline tests/fe/bdm_16.cc]: once you change it, it will be consistent ;-)

File: source/fe/fe_bdm.cc [hunk 5]
```diff
@@ -1,6 +1,6 @@
 // ---------------------------------------------------------------------
 //
-// Copyright (C) 2003 - 2016 by the deal.II authors
```
Comment (elddle) [inline source/fe/fe_bdm.cc]: I fixed this :)

**Comment by masterleinad** [comment]
@elddle Thanks again!
